### PR TITLE
CORE-1221: GET /api/v1/products/:id/permissions/me

### PR DESCRIPTION
**Pull request by harbu**
Fetch my permissions to a Product.

Used by front-end to check whether user can edit Product or not.

### CODE REVIEW

File: rest-e2e-tests/products-api.js [hunk 1]
```diff
@@ -12,6 +12,11 @@ const DEVOPS_USER_TOKEN = 'devops-user-key'
 const Streamr = initStreamrApi(URL, LOGGING_ENABLED)
```
Comment (kare) [inline rest-e2e-tests/products-api.js]: Rename tests to *.test.js

File: rest-e2e-tests/products-api.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ const DEVOPS_USER_TOKEN = 'devops-user-key'
...
 const schemaValidator = new SchemaValidator()
 
+function assertIsPermission(data) {
```
Comment (kare) [inline rest-e2e-tests/products-api.js]: % npm run test-rest-api

Fails with errors on my machine.

File: grails-app/conf/UrlMappings.groovy [hunk 3]
```diff
@@ -58,9 +59,12 @@ class UrlMappings {
 		"/api/v1/login/$action"(controller: "challengeApi")
 
 		"/api/v1/categories"(resources: "categoryApi")
+
 		"/api/v1/products"(resources: "productApi")
 		"/api/v1/products/$productId/streams"(resources: "productStreamsApi")
 		"/api/v1/products/$id/$action"(controller: "productApi")
+		"/api/v1/products/$resourceId/permissions/me"(controller: "permissionApi", action: "getOwnPermissions") { resourceClass = Product }
```
Comment (jtakalai) [inline grails-app/conf/UrlMappings.groovy]: is this a new thing? Looks kind of familiar.

File: rest-e2e-tests/schemas/permission.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
```
Comment (jtakalai) [inline rest-e2e-tests/schemas/permission.json]: where do these schemas go to? Can they be used with swagger or something?

File: rest-e2e-tests/products-api.js (hunk 2 above)
Comment (jtakalai) [reply rest-e2e-tests/products-api.js]: all tests pass for me

File: rest-e2e-tests/products-api.js (hunk 1 above)
Comment (harbu) [reply rest-e2e-tests/products-api.js]: If it's okay with you, I'd like to do the renaming after a couple of branches have been landed on marketplace-api. So then I can do it in a single swoop commit.

Comment (harbu) [reply rest-e2e-tests/products-api.js]: I've written a note for myself for this.

Comment (kare) [reply rest-e2e-tests/products-api.js]: 👍

File: rest-e2e-tests/products-api.js (hunk 2 above)
Comment (kare) [reply rest-e2e-tests/products-api.js]: All tests pass for me too!

**Pull request merged**